### PR TITLE
fix: allow using dot notation for `GroupFrom`/`MapFrom`

### LINKED ISSUE #44: allow using dot notation for `GroupFrom`/`MapFrom`
there are cases when dealing with api responses like where they return nested data and I just don't care about most of it and I'd like to get the nested data

e.g
`GroupFrom('contracts.registrant')`
or
`GroupFrom('contracts.registrant.name')`

### PR DESCRIPTION
**Pull request by webbrain-one**
Closes #44

### CODE REVIEW

File: src/Support/Data.php [hunk 1]
```diff
@@ -65,6 +66,27 @@ public static function wrap(string|array|object $data): self
 
     public function has(string $key): bool
     {
+        if (str_contains($key, '.')) {
```
Comment (ALameLlama) [inline src/Support/Data.php:69]: I think this logic needs to be moved down,

This is running before the existing isset check on $this->masked

e.g
```php

$data = Data::wrap([
    'a' => [
        'b' => [
            'c' => 'something'
        ],
    ],
]);

$data->unset('a');
echo $data->has('a'); // false
echo $data->has('a.b'); // true
```

Comment (ALameLlama) [inline src/Support/Data.php:69]: Actually this causes another issues since if you had a key named something like 'a.b' and we split right away it won't find it

I think this logic needs to be moved down,

This is running before the existing isset check on $this->masked

e.g
```php
$data = Data::wrap([
    'a.b' => 'something'
]);

echo $data->has('a.b'); // false
```

File: src/Support/Data.php [hunk 2]
```diff
@@ -65,6 +66,27 @@ public static function wrap(string|array|object $data): self
...
+            $source = is_array($this->originalData) ? $this->data : $this->originalData;
+            foreach (explode('.', $key) as $part) {
+                if (is_array($source) && array_key_exists($part, $source)) {
+                    $source = $source[$part];
+                } elseif (is_object($source) && property_exists($source, $part)) {
+                    $source = $source->{$part};
+                } elseif (is_object($source) && method_exists($source, '__get')) {
```
Comment (ALameLlama) [inline src/Support/Data.php:76]: this doesn't preserve the existing warning to exception we do on line 110 and 120

File: src/Support/Data.php [hunk 3]
```diff
@@ -118,6 +140,27 @@ public function has(string $key): bool
 
     public function get(string $key): mixed
     {
+        if (str_contains($key, '.')) {
```
Comment (ALameLlama) [inline src/Support/Data.php:143]: I think this has a lot of the same changed needed that the has() function has

**Review by ALameLlama — CHANGES_REQUESTED** [review]
I think this is a very good start but just needs to be tweaked a little to handle edge cases better, we'll also need to add tests to over all the cases to make sure all the existing data is working expected and please cover MapFrom and GroupFrom with array, object, and mixed nesting, plus masks/overrides, source consumption, terminal null, literal dotted keys, and magic properties
